util/util.go

```diff
@@ -137,6 +137,10 @@ func GetReceipts(ctx context.Context, rawBlocks []*json.RawMessage, c *ethrpc.Cl
 
 	var start uint64 = 0
 	for {
+		if len(blms) == 0 {
+			log.Debug().Int("Length of BatchElem", len(blms)).Msg("BatchElem is empty")
+			return nil, nil
+		}
```

Comment:
I think this check can be done before the `for` loop.

Reply:
Thanks for this!